tests/integration/src/all_data_types.rs

```diff
@@ -0,0 +1,174 @@
+use std::time::Duration;
+
+use anyhow::{Context, Result};
+use async_std::task;
+use fluvio_model_sql::{Operation, Upsert};
+use log::{debug, info};
+use serde::Deserialize;
+
+use crate::utils::{
+    self, ctx::TestContext, generate_records, new_config_path, produce_to_fluvio,
+    read_from_postgres,
+};
+
+pub async fn test_postgres_all_data_types(ctx: &mut TestContext) -> Result<()> {
+    #[derive(sqlx::FromRow, Debug)]
+    #[allow(dead_code)]
+    struct TestRecord {
+        bool_col: bool,
+        smallint_col: i16,
+        int_col: i32,
+        bigint_col: i64,
+        float_col: f32,
+        double_col: f64,
+        text_col: String,
+        bytes_col: Vec<u8>,
+        numeric_col: rust_decimal::Decimal,
+        timestamp_col: chrono::NaiveDateTime,
+        date_col: chrono::NaiveDate,
+        time_col: chrono::NaiveTime,
+        uuid_col: uuid::Uuid,
+        json_col: serde_json::Value,
+        char_col: i8,
+    }
+
+    info!("running 'test_postgres_all_data_types' test");
+    let config_path = new_config_path("test_postgres_all_data_types.yaml")?;
+    debug!("{config_path:?}");
+    let config: TestConfig = serde_yaml::from_reader(std::fs::File::open(&config_path)?)?;
+    let table = "test_postgres_all_data_types";
+    sqlx::query(&format!(
+        "CREATE TABLE {} (
```

Comment:
minor nit. move this as const to make it easier to read

Reply:
sure, what do you think of using static lazy?

```rust
const TABLE: &str = "test_postgres_all_data_types";
static CREATE_TABLE: Lazy<String> = Lazy::new(|| {
    format!(
        r#"
        CREATE TABLE {TABLE} (
            bool_col bool NOT NULL,
            smallint_col smallint NOT NULL,
            int_col int NOT NULL,
            bigint_col bigint NOT NULL,
            float_col float4 NOT NULL,
            double_col float8 NOT NULL,
            text_col varchar NOT NULL,
            bytes_col bytea NOT NULL,
            numeric_col numeric NOT NULL,
            timestamp_col timestamp NOT NULL,
            date_col date NOT NULL,
            time_col time NOT NULL,
            uuid_col uuid NOT NULL PRIMARY KEY,
            json_col json NOT NULL,
            char_col "char" NOT NULL
        )
        "#
    )
});
```

Reply:
or just write table directly and use const?

Reply:
Yes, const would work
```
const CREATE_TABLE: &str = #"
        CREATE TABLE {TABLE} (
            bool_col bool NOT NULL,
            smallint_col smallint NOT NULL,
            int_col int NOT NULL,
            bigint_col bigint NOT NULL,
            float_col float4 NOT NULL,
            double_col float8 NOT NULL,
            text_col varchar NOT NULL,
            bytes_col bytea NOT NULL,
            numeric_col numeric NOT NULL,
            timestamp_col timestamp NOT NULL,
            date_col date NOT NULL,
            time_col time NOT NULL,
            uuid_col uuid NOT NULL PRIMARY KEY,
            json_col json NOT NULL,
            char_col "char" NOT NULL
        )
        "#;
```

Reply:
I think that this "{TABLE}" variable call do not work for a const.